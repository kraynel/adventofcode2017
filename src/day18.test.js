const { executeProgram, executeInParallel } = require("./day18");
const fs = require("fs");

describe("executeProgram", () => {
  it("should get the transmitted value", () => {
    expect(
      executeProgram(`set a 1
add a 2
mul a a
mod a 5
snd a
set a 0
rcv a
jgz a -1
set a 1
jgz a -2`)
    ).toBe(4);
  });

  it("should get the value", done => {
    fs.readFile("inputs/day18.txt", "utf8", (err, data) => {
      expect(executeProgram(data)).toMatchSnapshot();
      done();
    });
  });
});

describe("executeProgram", () => {
  it("should get the transmitted value", () => {
    expect(
      executeInParallel(`snd 1
snd 2
snd p
rcv a
rcv b
rcv c
rcv d`)
    ).toBe(3);
  });

  it("should get the value", done => {
    fs.readFile("inputs/day18.txt", "utf8", (err, data) => {
      expect(executeInParallel(data)).toMatchSnapshot();
      done();
    });
  });
});
