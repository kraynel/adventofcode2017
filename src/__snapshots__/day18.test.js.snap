// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`executeProgram should get the value 1`] = `8600`;

exports[`executeProgram should get the value 2`] = `7239`;
